app/Services/MapboxGeocodingService.php

```diff
@@ -26,4 +29,35 @@ public function getCoordinatesFromApi(string $cityName, string $countryName): ar
             return [];
         }
     }
+
+    public function getPlaceFromApi(string $lat, string $long): array
+    {
+        $token = env("MAPBOX_TOKEN");
+        $client = new Client();
+
+        try {
+            $response = $client->get(
+                "https://api.mapbox.com/geocoding/v5/mapbox.places/$long,$lat.json?access_token=$token",
+            );
+            $features = json_decode($response->getBody()->getContents(), true)["features"];
+
+            foreach ($features as $key) {
+                $id = $key["id"];
+
+                if (str_contains($id, "place")) {
+                    $this->city = $key["text"];
+                } elseif (str_contains($id, "locality") && !isset($this->city)) {
+                    $this->city = $key["text"];
+                }
+
+                if (str_contains($id, "country")) {
+                    $this->country = $key["text"];
+                }
+            }
+
+            return [$this->city, $this->country];
+        } catch (Throwable) {
+            return [];
```

Comment:
Is there any use case to return empty array instead of throw an exception?

Throwing custom exception will be better in my point of view.
e.g. 
`throw new MapboxGeocodingServiceException(previous: $exception);` 
($exception var is from catch method)

Reply:
No, it was temporary solution
Added exception